### ISSUE TITLE
visible="no" ignored in treeview

### ISSUE THREAD

**Issue by h-2** [doxygen/doxygen#9617]
**Describe the bug**

I have a userpage within a usergroup and `visible="no"`. It correclty disappears from the index (thing at top), but it is still visible in the treeview.

Curiously, the usergroup disappears from the treeview if set to invisible. Its requested-as-invisible children then appear on the level above.

**Expected behavior**

it should be hidden

**Screenshots**

![image](https://user-images.githubusercontent.com/2996863/192579577-7763decf-074f-46dd-900c-9de9cb0ad61f.png)

**To Reproduce**

[bug.tar.gz](https://github.com/doxygen/doxygen/files/9657914/bug.tar.gz)

**Version**

```
1.9.5 (2f6875a5ca481a69a6f32650c77a667f87d25e88)
```

```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9632

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by h-2** [comment]
Thanks a lot!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
